src/shared/workers/index.ts

```diff
@@ -7,7 +7,9 @@ export enum WorkerScript {
 export function createWorker(scriptName: WorkerScript) {
   const worker = new Worker(scriptName);
   worker.addEventListener('error', error => {
-    void analytics.track(`worker_error_thrown_${scriptName}`, { error });
+    void getAnalyticsClient().then(
+      client => client && client.track(`worker_error_thrown_${scriptName}`, { error })
```

Comment:
This API seems much clunkier than before. What's the benefit wrapping it in a promise, vs how it worked before? Ideally, we wouldn't have to check for the existence of the client every time we want to track an analytics event.

Check the official docs example, which doesn't require resolving a promise on every use https://github.com/segmentio/analytics-next#using-react-simple